### PR TITLE
Use Thor for built-in middleware task

### PR DESCRIPTION
**Pull request by p8**
Currently we use both Thor and Rake for `bin/rails` commands.
We eventually want to get all the built-in tasks promoted to Thor Commands.

This migrates the `middleware` task to Thor.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: railties/test/commands/middleware_test.rb [hunk 1]
```diff
@@ -0,0 +1,427 @@
+# frozen_string_literal: true
+
+require "isolation/abstract_unit"
+require "rails/command"
+require "rails/commands/middleware/middleware_command"
```
Comment (jonathanhefner) [inline railties/test/commands/middleware_test.rb:5]: I think we can drop this `require`.

```suggestion
```

**Review by jonathanhefner — COMMENTED** [review]
One minor nit, but looks good! :+1:

File: railties/test/commands/middleware_test.rb (hunk 1 above)
Comment (p8) [reply railties/test/commands/middleware_test.rb]: Thanks! I've removed it.
This pattern might be used in other tests as well.
I'll see if I can remove some more requires in another PR.

**Comment by jonathanhefner** [comment]
Thank you, @p8! :sandwich:

**Pull request merged**